src/copa_modules/_data_processor/data_processor.py

```diff
@@ -0,0 +1,314 @@
+import os
+from typing import Dict, List, Union
+
+import pandas as pd
+from halo import Halo
+
+from .._error import CopaError
+from .._logger import Logger
+from .._types import data_config, dataset, file_types
+
+
+class data_processor:
+    """
+    The DataProcessor class is responsible for processing data based on a given configuration.
+
+    Attributes:
+        config (DataConfig): The configuration object containing parameters for data processing.
+        base_path (str): The base path where the data is located.
+        data (Dataset): The dataset object that will hold the processed data, initialized as an empty dictionary.
+    """
+
+    def __init__(
+        self,
+        config: data_config | None,
+        base_path: str = "",
+        test_mode: bool = False,
+        test_rows: int = 100,
+        drop_duplicates: bool = False,
+        no_cache: bool = False,
+    ):
+        """
+        Initializes the DataProcessor object with the provided configuration and options.
+
+        Args:
+            config (DataConfig | None): The configuration object containing parameters for data processing.
+                                        If None, a CopaError will be raised.
+            base_path (str, optional): The base path where the data is located. Defaults to an empty string,
+                                        which means the data is located in the current directory.
+            test_mode (bool, optional): If True, the DataProcessor will run in test mode. Defaults to False.
+            test_rows (int, optional): The number of rows to process in test mode. Defaults to 100.
```

Comment:
I I run in test mode with 100 rows, a cached file is created. If I rerun with a different number of rows, is the cache automatically overwritten? Or is the original cache used (in which case the number of rows will not be correct.)

Reply:
I have updated the readme accordingly.